### PR TITLE
Total rewrite

### PR DESCRIPTION
**Pull request by vmarchaud**

### CODE REVIEW

**Comment by vmarchaud** [comment]
@f-hj Ready to be tested on windows + macos

File: src/actions.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        })
+      })
+
+      io.action('vmstats', function (reply) {
+        exec('vmstat -S m', function (err, out) {
+          return err ? reply(err.message) : reply(out.replace(/\n/g, '<br />'))
+        })
+      })
+    }
+
```
Comment (f-hj) [inline src/actions.js]:
```
if (process.platform !== 'win32' && process.platform !== '')
```

File: src/metrics/processes.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  fetchDarwin () {
+    exec('ps -A', (err, stdout) => {
+      if (err || stdout.length === 0) return console.error(`Failed to retrieve process count for darwin`, err)
+      this.processCount.set(stdout.split('\n').length - 1)
+    })
+  }
+
+  fetch () {
```
Comment (f-hj) [inline src/metrics/processes.js]: On win, doesn't crash but make logs (same for all unix commands)

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    block: {
+      actions: false,
+      issues: true,
+      meta: true,
+      cpu: false,
+      mem: false,
+      main_probes: [
+        'CPU usage',
+        'Free memory',
+        'Avail. Disk',
```
Comment (f-hj) [inline src/index.js]: Display even if it's not available on current platform?

File: src/actions.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      })
+    })
+
+    io.action('open ports', function (reply) {
+      exec('lsof -Pni4 | grep ESTABLISHED', function (err, out) {
+        return err ? reply(err.message) : reply(out.replace(/\n/g, '<br />'))
+      })
+    })
+
+    io.action('ifconfig', function (reply) {
```
Comment (f-hj) [inline src/actions.js]: `ipconfig` on win

File: src/metrics/processes.js (hunk 2 above)
Comment (vmarchaud) [reply src/metrics/processes.js]: Expected, i think its okay to log errors

File: src/index.js (hunk 3 above)
Comment (vmarchaud) [reply src/index.js]: it's only used for keymetrics so it's not really important for me

File: src/actions.js (hunk 4 above)
Comment (vmarchaud) [reply src/actions.js]: :+1:

File: src/actions.js (hunk 1 above)
Comment (vmarchaud) [reply src/actions.js]: :+1:

**Pull request merged**